lib/babel.js

```diff
@@ -74,8 +77,33 @@ if (inputSourceMap) {
 	options.inputSourceMap = JSON.parse(inputSourceMap.map);
 }
 
+// load up custom compilers
+process.argv.slice(3).forEach(function (arg) {
+	var match = /^--compilers=.+?:(.+)$/i.exec(arg);
+	if (match) {
+		require(match[1]);
+	}
+});
+
 // include test file
-var transpiled = babel.transformFileSync(testPath, options);
+var Module = module.constructor;
+var source = fs.readFileSync(testPath, 'utf8');
+var extension = path.extname(testPath);
+
+var compiler = Module._extensions[extension];
+if (compiler) {
+	compiler(
```

Comment:
Interestingly even without the `--compiler` flag being used for `.js` extensions this would load the default compiler. You could fall back to the `.js` compiler and never `readFileSync` the source above.